### ISSUE TITLE
iRacing Porsche 919

### ISSUE THREAD

**Issue by andreasdahl1987** [SHWotever/SimHub#670]
Hi! 

Request regarding this car: 
In-game you can toggle the TC and NoBoost. There are a few properties with names that suggest they would give some info on these toggles, however they dont. There is no in-game indication wether you've turned of the hybrid system or the TC, it can only be experienced by driving. Would be really helpful to be able to display this on my simhub dash. Could you have a look if this telemetry is available from the game? 

If you dont have the Porsche 919, I could make a donation corresponding to the cost of the car.

Kind regards,
Andreas

**Comment by SHWotever** [comment]
Hi ! 

The available telemetry entries will self update when you will run for the first time with the car, but indeed the iracing model is quite painful, each car exposes unique telemetry data.

Can you send me a replay with this particular car ?
You simply have to press record, take a short lap, press live to stop the record,
![image](https://user-images.githubusercontent.com/2207331/94535200-91f7d480-0241-11eb-9f50-cdbaefb551fe.png)

The the record files will be stored here : c:\Program Files (x86)\SimHub\Replays\iracing, you simply have to send it to me (using any file sharing service of your choice, mediafire, dropbox, googledrive ...)

Nicolas

**Comment by Romainrob** [comment]
Hi. Just to throw my 2 cents.
To my knowledge, these are broken on the iRacing side of things. At least they used to be.
I remember looking into this and reading complaints on the forums about it.
They indeed showed up in the SimHub property list, but telemetry.dc.HysNoBoostToggle and telemetry.ManualNoBoost did nothing and dc.TractionControlToggle weirdly varied from car to cars.
Obviously they may have been fixed since then but they never mentioned it in the build/patch release note...
Have a nice day!

**Comment by andreasdahl1987** [comment]
Any updates?

**Comment by SHWotever** [comment]
Hello ! 

sorry I missed your reply ;) I will check after work ;)

**Comment by SHWotever** [comment]
Hello !

Sorry for the long delay, I can confirm the telemetry contains no interesting data about noboost,

The telemetry contains the 4 following :
![image](https://user-images.githubusercontent.com/2207331/97145365-6398e680-1766-11eb-9505-5054e990beb8.png)

But unfortunately I could only see activity for manualboost.

I'm affraid that nothing can be done on my end, but as soon as Iracing will get it fixed it will work.

Nicolas

**Comment by andreasdahl1987** [comment]
Thanks for looking at this! 

To follow up; I did some experimenting with the 919, Mclaren MP4-30 and Dallara P217 to figure out how this TC toggle and noBoostHold works:

TC toggle : 5 second deactivation of traction control. Can't be turned back on in that period; if you press again you reset the timer to 5 seconds. Works both in speed and standing still. Works the same way for 919 and P217. Useful for flicking the car around if you spin off the road 180. I've made a script to return this toggle on/off and a timer. Works like a charm :)

NoBoostToggle: Simply turns off the hybrid system. Cant turned off in pits. Will turn back on if you go below 80 kph or SoC is 100%. Also made a script to track this parameter, works perfectly :)
